### ISSUE TITLE
False negative: ExpressionOverBlockLambda silently passes block lambdas in switch rule

### ISSUE THREAD

**Issue by lithops-zty** [checkstyle/checkstyle#21515]
I have read check documentation: https://checkstyle.org/checks/coding/expressionoverblocklambda.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
/var/tmp $ javac ControlLambda.java
/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="ExpressionOverBlockLambda"/>
  </module>
</module>
/var/tmp $ cat ControlLambda.java
import java.util.function.Supplier;
public class ControlLambda {
    Supplier<String> c1 = () -> { return "ctrl"; };  // flagged

    Supplier<String> rule(int x) {
        return switch (x) {
            case 1 -> () -> { return "one"; };  // should be flagged too
            default -> () -> "other";
        };
    }
}
/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-14.1.0-all.jar -c config.xml ControlLambda.java
Starting audit...
[ERROR] /var/tmp/ControlLambda.java:4:30: Expression lambdas are preferred over single-line block lambdas. [ExpressionOverBlockLambda]
Audit done.
Checkstyle ends with 1 errors.
```

---

**Describe what you expect in detail.**

Both `c1` and the `case 1` rule value are single-line block lambdas whose only statement is `return <expr>;` and are convertible to expression lambdas (`() -> "ctrl"` / `() -> "one"`). The check should report **both**. Only line 4 is reported; the switch-rule value lambda on line 9 is silently skipped.

**Comment by sahitya0xsingh** [comment]
I am on it.
